### ISSUE TITLE
Main "open initiative" button should always open something (even if URL isn't configured)

### ISSUE THREAD

**Issue by ohadschn** [4tals/LinksForIsrael#415]
For example: https://github.com/4tals/LinksForIsrael/blob/e56214883eab6b5e59c826de7ec779127cfd24c8/_data/links/GeneralAssistance/links.json#L6C15-L6C15

Since the url property is not defined, the main button doesn't do anything (you must click the WhatsApp icon):
![image](https://github.com/4tals/LinksForIsrael/assets/8555641/e2055e52-7db2-42ad-80b8-f555806545d7)

We need to fall back to one of the links that do exist, at least until we enforce mandatory properties with https://github.com/4tals/LinksForIsrael/issues/293

**Comment by ohadschn** [comment]
> but if this initiative doesn't have a website? why don't we just remove this button if there is not initiative to open? IMO mandatory fields should be:
> 
> 1. title
> 2. informative description
> 3. at least one of: whatsapp, linkedin, website, etc...

As I see it, that button doesn't mean "initiative website" as much as it means "main initiative link"
If the initiative is a WhatsApp group, that button should open the WhatsApp link

**Comment by 4tal** [comment]
It's not on the frontend level, maybe they can add warning sign next to initiative, that with hover, say that link.url is empty. but it should be done on validation level. link url is the exact purpose of been the main link.

**Comment by ohadschn** [comment]
> It's not on the frontend level, maybe they can add warning sign next to initiative, that with hover, say that link.url is empty. but it should be done on validation level. link url is the exact purpose of been the main link.
> 
> 1. Add to the PR validation check that link.url is not empty (I think there is).
> 2. Add build check if exists link without URL.

I agree this is a workaround, which is why I stated in the issue (bottom line, below the image):

> We need to fall back to one of the links that do exist, at least until we enforce mandatory properties with https://github.com/4tals/LinksForIsrael/issues/293

Yes, we could add validation for this specific field, but I think we might as well do an easy fallback on the frontend side until we have the proper schema validation at the PR validation level

**Comment by 4tal** [comment]
Got It. Agree

**Comment by omer-friedman** [comment]
so what are the conclusions?
fallback? to where?

**Comment by ohadschn** [comment]
> so what are the conclusions? fallback? to where?

Fallback by whatever order like
1. Website
2. Facebook
3. Instagram
4. WhatsApp
5. whatever...

**Comment by 4tal** [comment]
For the main button try to assign url, if not url, website, than, whatsapp, facebook, instagram and the order of the rest is not that important just Or operator.
